### PR TITLE
[base] Make Ginger search open for extension

### PR DESCRIPTION
**Pull request by ddeboer**
* Replace call to search_query:search with notification.
* Make sure to remove all custom search arguments, as Zotonic
  will trip over them.

### CODE REVIEW

File: modules/mod_ginger_base/support/ginger_search.erl [hunk 1]
```diff
@@ -10,20 +10,37 @@
...
+            lists:append([Args, [{filters, Filters}]])
     end,
 
     QueryArgs = merge_ginger_args(Args1, Context),
-    search_query:search(QueryArgs, Context).
+    Query = #search_query{search = {'query', QueryArgs}, offsetlimit = Offset},
+
+    %% Forward to the next observer. Make sure all custom Ginger search
+    %% arguments have been removed.
+    z_notifier:first(Query, Context).
```
Comment (ddeboer) [inline modules/mod_ginger_base/support/ginger_search.erl]: @Dirklectisch Was there any compelling reason *not* to do this in the first place? After removing the custom search arguments, forwarding to Zotonic’s search (or in my case, Elasticsearch) seems to work well.

File: modules/mod_ginger_base/support/ginger_search.erl [hunk 2]
```diff
@@ -10,20 +10,37 @@
...
         undefined ->
-            Args1 = Args;
+            Args;
         Filters ->
-            Args1 = lists:append([Args, [{filters, Filters}]])
+            lists:append([Args, [{filters, Filters}]])
     end,
 
     QueryArgs = merge_ginger_args(Args1, Context),
-    search_query:search(QueryArgs, Context).
```
Comment (ddeboer) [inline modules/mod_ginger_base/support/ginger_search.erl]: This blocked the chain of observers, which is almost always a bad thing as it makes the code [closed for extension](https://en.wikipedia.org/wiki/Open/closed_principle).

**Comment by ddeboer** [comment]
Adapted with a quick solution to remain compatible with ginger_geo_search. I considered refactoring of the Zotonic search notifications (split `search_query` into two notifications, as discussed verbally) out of scope for this PR.

**Comment by Dirklectisch** [comment]
I tested this on one of our sites that has a map on the homepage and it seems to work fine.

**Pull request merged**